examples/source_examples/accelo_api_connector_multithreading_enabled/connector.py

```diff
@@ -0,0 +1,760 @@
+"""
+Accelo API Connector for Fivetran
+
+This module implements a connector for syncing data from the Accelo API to Fivetran.
+It handles authentication, rate limiting, and data synchronization for companies,
+invoices, payments, prospects, jobs, and staff.
+
+Author: Ahmed Zedan
+Date: 2024-09-20
+Version: 1.3
+"""
+import json
+from fivetran_connector_sdk import Connector, Logging as log, Operations as op
+import requests
+import time
+from datetime import datetime, timezone
+from concurrent.futures import ThreadPoolExecutor
+from threading import local
+from copy import deepcopy
+
+from dotenv import load_dotenv
+
+# Load environment variables from .env file
+load_dotenv()
+
+# Constants
+BASE_URL = "https://{deployment}.api.accelo.com/api/v0"
+RATE_LIMIT = 5000  # Requests per hour
+RATE_LIMIT_INTERVAL = 3600  # 1 hour in seconds
+CHECKPOINT_INTERVAL = 1000  # Number of records to process before creating a checkpoint
+MAX_WORKERS = 5  # Number of concurrent API requests
+REQUEST_TIMEOUT = 30  # Default timeout in seconds
+SYNC_TIMEOUT = 3600  # Timeout for the entire sync operation in seconds
+BATCH_SIZE = 100  # Maximum number of records per page in Accelo API (API limit)
+RETRIES = 3
+
+# Create a thread-local state object
+thread_local_state = local()
+
+
+def get_access_token(client_id, client_secret, deployment):
+    """
+    Obtain an access token from the Accelo API using OAuth 2.0 client credentials flow.
```

Comment:
And we also have an example on using OAuth to get data from a source.

we should just reference this example rather than duplicate - DRY.

Reply:
I don't see any OAuth e.g. currently..